test/e2e/targets/azureeventhubs/main.go

```diff
@@ -190,12 +191,105 @@ var _ = Describe("Azure Event Hubs target", func() {
 
 				By("verifying the sent event", func() {
 					Expect(len(payload)).To(BeNumerically(">", 0))
+					var ce cloudevents.Event
 
-					// NOTE: The payload will be a stringified version of the CloudEvent
-					Expect(payload).To(ContainSubstring(string(event.Data())))
-					Expect(payload).To(ContainSubstring("type: " + event.Type()))
-					Expect(payload).To(ContainSubstring("source: " + event.Source()))
-					Expect(payload).To(ContainSubstring("id: " + event.ID()))
+					err := json.Unmarshal(payload, &ce)
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(ce.Data()).To(ContainSubstring(string(event.Data())))
```

Comment:
I missed that one earlier, but it should be an `Equal` here.
The adapter isn't supposed to alter the events' data ("Hello, World!") while serializing it to raw bytes.

https://github.com/triggermesh/triggermesh/blob/12b2f1e86ba081390d82f1db852059777a0e1159/test/e2e/targets/awss3/main.go#L169

Reply:
Thought I got that one too.